src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,145 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newValue = new Node<>(last, value, null);
+        if (isEmpty()) {
+            last = newValue;
+            first = newValue;
+        } else {
+            last.next = newValue;
+            newValue.prev = last;
+            last = newValue;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+        Node<T> newValue = new Node<>(null, value, null);
+        if (isEmpty()) {
+            last = newValue;
+            first = newValue;
+        } else if (index == 0) {
+            newValue.next = first;
+            first = newValue;
+        } else if (index == size) {
+            last.next = newValue;
+            newValue.prev = last;
+            last = newValue;
+        } else {
+            Node<T> current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            newValue.prev = current.prev;
+            current.prev.next = newValue;
+            newValue.next = current;
+            current.prev = newValue;
```

Comment:
Also you can split this code in methods to add nodes at first position and before

Reply:
I think that in this case it is not necessary to do it